orchagent/portsorch.cpp

```diff
@@ -5612,6 +5612,10 @@ void PortsOrch::doTask(NotificationConsumer &consumer)
                     updateDbPortOperSpeed(port, speed);
                 }
             }
+            else
+            {
+                updateDbPortOperSpeed(port, 0);
+            }
```

Comment:
i think we should update speed=0 only when the operational status is UP similar to done in the failure case [here](https://github.com/Azure/sonic-swss/pull/2119/files#diff-bb736c6a11da68afacf95d761fb52a3c11726cdd72c967d4803b8c9b1a6dcb69R5728)